src/components/OptionSwitcher.vue

```diff
@@ -0,0 +1,81 @@
+<script setup lang="ts">
+import { ref } from 'vue'
+
+const props = defineProps({
+  items: { type: Array<{ name: string }>, required: true },
+  modelView: { type: String, required: true }
+})
+
+const selected = ref(props.modelView)
+</script>
```

Comment:
You had better use computed properties here instead of ref.
https://vuejs.org/guide/essentials/computed.html#computed-properties